eng/pipelines/libraries/run-test-job.yml

```diff
@@ -60,7 +61,7 @@ jobs:
         - ${{ if notIn(parameters.framework, 'allConfigurations', 'net48') }}:
           - ${{ format('libraries_build_{0}{1}_{2}_{3}', parameters.osGroup, parameters.osSubgroup, parameters.archType, parameters.buildConfig) }}
           # tests are built as part of product build
-          - ${{ if or(ne(parameters.archType, parameters.dependsOnTestArchitecture), ne(parameters.buildConfig, parameters.dependsOnTestBuildConfiguration)) }}:
+          - ${{ if or(ne(parameters.archType, parameters.dependsOnTestArchitecture), ne(parameters.buildConfig, parameters.dependsOnTestBuildConfiguration), ne(parameters.osSubgroup, parameters.dependsOnTestOsSubgroup)) }}:
             - ${{ format('libraries_build_{0}_{1}_{2}', parameters.osGroup, parameters.dependsOnTestArchitecture, parameters.dependsOnTestBuildConfiguration) }}
```

Comment:
Shouldn't you specify {0}{1} for parameters.osGroup / parameters.dependsOnOsSubgroup here? I understand that right now dependsOnOsSubgroup is always empty as you haven't changed any of the callers but the first time anyone tries to use it for depending on a non-trivial OS subgroup I suspect this will blow up.

Reply:
Yeah, I was actually just pushing that. Good catch.